velox/connectors/hive/storage_adapters/s3fs/RegisterS3FileSystem.cpp

```diff
@@ -52,12 +53,36 @@ fileSystemGenerator() {
   };
   return filesystemGenerator;
 }
+
+std::function<std::unique_ptr<velox::dwio::common::FileSink>(
+    const std::string&,
+    const velox::dwio::common::FileSink::Options& options)>
+s3WriteFileSinkGenerator() {
+  static auto s3WriteFileSink =
+      [](const std::string& fileURI,
+         const velox::dwio::common::FileSink::Options& options) {
+        if (isS3File(fileURI)) {
+          auto fileSystem =
+              filesystems::getFileSystem(fileURI, options.connectorProperties);
+          return std::make_unique<dwio::common::WriteFileSink>(
+              fileSystem->openFileForWrite(fileURI),
+              fileURI,
+              options.metricLogger,
+              options.stats);
+        }
+        return static_cast<std::unique_ptr<dwio::common::WriteFileSink>>(
```

Comment:
are you doing this just because of lambda return type inference? If so, can you just return nullptr here, and enforce the lambda return type at its definition?

Reply:
fixed.